project/src/main/java/com/google/sps/data/FakePlaceDetailsRequest.java

```diff
@@ -0,0 +1,21 @@
+package com.google.sps.data;
+
+import com.google.maps.PlaceDetailsRequest;
+import com.google.maps.GeoApiContext;
+
+public class FakePlaceDetailsRequest extends PlaceDetailsRequest {
+
+    /** The Place ID to retrieve details for. */
+    public String placeId;
+
+    /**
+     * FakeTestSearchRequest constructor.
+     *
+     * @param geoApiContext the GeoApiContext used for all Google GEO API requests
+     * @param placeId the Place ID to retrieve details for
+     */
+    public FakePlaceDetailsRequest(GeoApiContext context, String placeId) {
+        super(context);
```

Comment:
Why do you need the context here?

Reply:
Because the parent's constructor takes this as a parameter (it has no default parameter-less constructor), so I need to call super() with this argument. I did wonder if I should create some dummy context to pass here instead. What do you think?

Reply:
Oh I forgot that in Java you have to call super... OK so it's fine as-is.